pkg/plugins/runtime/k8s/controllers/pod_converter.go

```diff
@@ -270,9 +272,20 @@ func (p *PodConverter) dataplaneFor(
 			return nil, errors.Errorf("invalid delegated gateway type '%s'", gwType)
 		}
 	} else {
-		ifaces, err := p.InboundConverter.InboundInterfacesFor(ctx, p.Zone, pod, services)
-		if err != nil {
-			return nil, err
+		var ifaces []*mesh_proto.Dataplane_Networking_Inbound
+		var err error
+		// deduplicate inbounds only when MeshService mode is exclusive. Without MeshServices we need duplicated inbounds
+		// for identity and traffic routing
```

Comment:
how identity is affected? it's still based on `kuma.io/service` even with MeshServices no?

Reply:
We base identity on tags, and we are basically removing identities by deduplicating inbounds and removing inbound and its tag kuma.io/service